sequencer/src/api.rs

```diff
@@ -1,26 +1,30 @@
-use crate::{network, Node, SeqTypes};
+use crate::{enriched_handle::EnrichedSystemContextHandle, network, Node, SeqTypes};
 use async_std::task::JoinHandle;
 use data_source::SubmitDataSource;
-use hotshot::types::SystemContextHandle;
 use hotshot_query_service::data_source::ExtensibleDataSource;
 
 mod data_source;
 pub mod endpoints;
 pub mod fs;
 pub mod options;
+pub mod signature_pool;
 pub mod sql;
 mod update;
 
+use hotshot_types::light_client::StateKeyPair;
 pub use options::Options;
 
+use self::data_source::StateSignatureDataSource;
+
 type NodeIndex = u64;
 
-pub type Consensus<N> = SystemContextHandle<SeqTypes, Node<N>>;
+pub type Consensus<N> = EnrichedSystemContextHandle<SeqTypes, Node<N>>;
 
 pub struct SequencerNode<N: network::Type> {
```

Comment:
I feel like `EnrichedSystemContextHandle` and `SequencerNode` are both kind of serving the same purpose: wrapping a `SystemContextHandle` with some extra state. What do you think about flattening it out and adding the signature pool directly to `SequencerNode`?

Reply:
Sure that'll be the best.
I was a little frustrated by the current initialization workflow. Let me spend more time to find a way.